server/src/main/java/org/opensearch/cluster/service/MasterService.java

```diff
@@ -131,6 +143,8 @@ public MasterService(Settings settings, ClusterSettings clusterSettings, ThreadP
             this::setSlowTaskLoggingThreshold
         );
 
+        this.throttlingStats = new MasterThrottlingStats();
+        this.masterTaskThrottler = new MasterTaskThrottler(clusterSettings, this, throttlingStats);
```

Comment:
Please avoid passing the complete reference of `MasterService ` to `MasterTaskThrottler ` I think we just need a mechanism to get the node version, use a `Supplier` instead

Reply:
Used `Supplier` for min Node version instead of passing `MasterService`.